src/Firestore.elm

```diff
@@ -189,21 +189,64 @@ create fieldDecoder encoder (Firestore config path) =
         }
 
 
+{-| Creates a document with a given document id.
+Takes the document id as the first argument.
+-}
+create : String -> FSDecode.Decoder a -> FSEncode.Encoder -> Firestore -> Task.Task Error (Document a)
+create documentId fieldDecoder encoder (Firestore config path) =
+    Http.task
+        { method = "POST"
+        , headers = Config.httpHeader config
+        , url =
+            Config.endpoint
+                [ UrlBuilder.string "documentId" documentId ]
+                path
+                config
+        , body = Http.jsonBody <| FSEncode.encode encoder
+        , timeout = Nothing
+        , resolver =
+            fieldDecoder
+                |> Document.decodeOne
+                |> jsonResolver
+        }
+
+
+{-| Overrides an existing document. Creates one if not present.
+-}
+override : FSDecode.Decoder a -> FSEncode.Encoder -> Firestore -> Task.Task Error (Document a)
+override fieldDecoder encoder (Firestore config path) =
```

Comment:
How about naming this as `upsert`?  We now have `insert` and this function also has functionality creating a new document not just updating, so I think it will sound more familier.